smarty/sampler.py

```diff
@@ -131,54 +144,58 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
+        #    # If this type is used, then track it
+        #    if basetype_typecounts[atom_type] > 0:
+        #       used_basetypes.append( ( smarts, atom_type) )
+        #    # If unused, it will be removed and the smarts stored in the no match list
+        #    else:
+        #        self.atomtypes_with_no_matches.add( smarts )
+        #        if self.verbose: print("Removing basetype '%s' ('%s'), which is unused." % (smarts, atom_type))
+        ## Atom basetypes to create new smart strings
+        #self.basetypes = copy.deepcopy(used_basetypes)
+        
+        print "typing..."
 
         # Type all molecules with current typelist to ensure that starting types are sufficient.
         self.type_molecules(self.atomtypes, self.molecules, self.element)
+
+        print "computing..."
+
         # Compute atomtype statistics on molecules for current atomtype set
         [atom_typecounts, molecule_typecounts] = self.compute_type_statistics(self.atomtypes, self.molecules, self.element)
         if self.verbose:
             print("MATCHED INITIAL TYPES:")
             self.show_type_statistics(self.atomtypes, atom_typecounts, molecule_typecounts)
 
-        # Track only used atomtypes and add unused to atomtypes with no matches
-        used_initial_atomtypes = list()
-        for (smarts, atom_type) in self.atomtypes:
-            if atom_typecounts[atom_type] > 0:
-                used_initial_atomtypes.append( (smarts, atom_type) )
-            else:
-                self.atomtypes_with_no_matches.add( smarts )
-                if self.verbose: print("Removing initial atom type '%s', as it matches no atoms" % smarts)
-        self.atomtypes = copy.deepcopy(used_initial_atomtypes)
-        self.initial_atomtypes = copy.deepcopy(used_initial_atomtypes)
+        ## Track only used atomtypes and add unused to atomtypes with no matches
+        #used_initial_atomtypes = list()
+        #for (smarts, atom_type) in self.atomtypes:
+        #    if atom_typecounts[atom_type] > 0:
+        #        used_initial_atomtypes.append( (smarts, atom_type) )
+        #    else:
+        #        self.atomtypes_with_no_matches.add( smarts )
+        #        if self.verbose: print("Removing initial atom type '%s', as it matches no atoms" % smarts)
+        #self.atomtypes = copy.deepcopy(used_initial_atomtypes)
+        #self.initial_atomtypes = copy.deepcopy(used_initial_atomtypes)
 
-        # Type molecules again with the updated atomtype list
-        self.type_molecules(self.atomtypes, self.molecules, self.element)
+        ## Type molecules again with the updated atomtype list
+        #self.type_molecules(self.atomtypes, self.molecules, self.element)
 
-        # These are atomtypes where not all children have been matched
-        self.unmatched_atomtypes = copy.deepcopy(self.atomtypes)
+        ## These are atomtypes where not all children have been matched
+        #self.unmatched_atomtypes = copy.deepcopy(self.atomtypes)
```

Comment:
@camizanette Why is the initial types section all commented out? It should still behave the same way as before.